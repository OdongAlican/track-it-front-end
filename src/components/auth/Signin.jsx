/* eslint-disable react/no-unescaped-entities */
/* eslint-disable react/require-default-props */

import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { signInAction } from '../../actions';

const Signin = props => {
  const { history } = props;
  const dispatch = useDispatch();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const saveData = event => {
    event.preventDefault();
    dispatch(signInAction({ username, password }, history));
  };

  const errorMessage = useSelector(state => state.authReducer.error);

  return (
    <div data-testid="appSignin" className="general-login-section">
      <div className="login-section p-3">
        <h3>Log In</h3>
        <form onSubmit={saveData}>
          <input placeholder="Enter User Name" value={username} onChange={e => setUsername(e.target.value)} />
          <input placeholder="Enter Password" value={password} onChange={e => setPassword(e.target.value)} type="password" />
          <p className="text-danger">{ errorMessage || '' }</p>
          <button type="submit" className="btn btn-primary">Submit</button>
        </form>
        <Link to={{
          pathname: '/signup',
        }}
        >
          Don't have an account?
        </Link>
      </div>

    </div>
  );
};

Signin.propTypes = {
  history: PropTypes.string,
};

export default Signin;
