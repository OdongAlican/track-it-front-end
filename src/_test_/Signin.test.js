import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Signin from '../components/auth/Signin';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};
const mockStore = configureStore();
let store;
const props = {
  history: '/',
};

it('renders Signin correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signin props={props} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSignin');
  expect(result[0]).toHaveTextContent("Log IntestSubmitDon't have an account?");
});

it('renders Signin correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signin props={props} /></Provider>
    </Router>,
  );
  const result = getAllByTestId('appSignin');
  expect(result[0]).not.toHaveTextContent('Some Random Text');
});

it('matches Signin snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Signin props={props} /></Provider>
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
