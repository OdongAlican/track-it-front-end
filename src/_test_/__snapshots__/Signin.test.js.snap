// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Signin snapshot 1`] = `
<div
  className="general-login-section"
  data-testid="appSignin"
>
  <div
    className="login-section p-3"
  >
    <h3>
      Log In
    </h3>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Enter User Name"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <p
        className="text-danger"
      >
        test
      </p>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <a
      href="/signup"
      onClick={[Function]}
    >
      Don't have an account?
    </a>
  </div>
</div>
`;
